ocfweb/account/vhost_mail.py

```diff
@@ -69,12 +76,13 @@ def vhost_mail(request):
 @login_required
 @group_account_required
 @require_POST
-def vhost_mail_update(request):
+def vhost_mail_update(request: HttpRequest) -> HttpResponseRedirect:
     user = logged_in_user(request)
 
     # All requests are required to have these
     action = _get_action(request)
-    addr_name, addr_domain, addr_vhost = _get_addr(request, user, 'addr', required=True)
+    # _get_addr should either return a valid tuple or error so it's not necessary to verify the Optional type
```

Comment:
You can annotate `_error` as `def _error(...) -> NoReturn`, and then remove the `Optional` from `_get_addr` (also you should probably do the same for anything that calls `_error`). 

In general if you want to unwrap an `Optional` that you know will never be `None`, you can do something like:

```
addr = _get_addr(...)
assert addr is not None
addr_name, addr_domain, addr_vhost = addr
```

Reply:
`Optional` can't be removed from `_get_addr` because it could return `None`, but just not for that particular call. The `assert` statement works well, though. Thanks for the suggestions 😃

Reply:
It cannot actually, all of the cases where it doesn't return are calls to `_error`, so it is really `Union[NoReturn, ...]`, but I think it is fine to leave this as is.

Reply:
It does return `None` sometimes- I tested it by putting `assert False` at the bottom of the function instead of `return None` and it does raise an AssertionError.. So maybe `Optional[Union[NoReturn, Tuple...]]` might be preferred?

Reply:
Oh yeah sorry if `original` is None and required is false then it returns None. I think it is fine as is.